sqlitecluster/SQLite.cpp

```diff
@@ -401,10 +401,13 @@ bool SQLite::beginTransaction(TRANSACTION_TYPE type) {
     // Reset before the query, as it's possible the query sets these.
     _autoRolledBack = false;
 
-    SINFO("[concurrent] Beginning transaction");
+    SINFO("[concurrent] Beginning transaction - open transaction count: " << (_sharedData.openTransactionCount + 1));
     uint64_t before = STimeNow();
     _insideTransaction = !SQuery(_db, "starting db transaction", "BEGIN CONCURRENT");
 
+    // We actively track transaction counts incrementing and decrementing to log the number of active open transactions at any given moment.
+    _sharedData.openTransactionCount++;
```

Comment:
Move this up above the logline so we don't need to add one in the logline?

Whether we should count the number of open transactions before/after we run the query is indeterminate, there seems as good as here.

Could also do `SINFO("[concurrent] Beginning transaction - open transaction count: " << ++_sharedData.openTransactionCount);` And the increment is atomic but it's not actually atomic with what the DB is doing, but I don't think it matters that much.

I'd be inclined to update the count of transactions before the query runs rather than after, and over rather than underestimate transactions.

Reply:
👍